sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1167,6 +1187,73 @@ Exceptions:
   `property::graph::updatable` was not set when the executable graph was created.
 * Throws with error code `invalid` if any node in `nodes` is not part of the
   graph.
+
+|
+[source, c++]
+----
+void
+update(const command_graph<graph_state::modifiable>& graph);
+----
+
+|Updates the executable graph node inputs & outputs from a topologically
+identical modifiable graph. A topologically identical graph is one with the
+same structure of nodes and edges. Additionally, the nodes and edges must have
+been added in the same order to both graphs.
```

Comment:
I feel like same order could be confusing. Should we clarify that this is not the order of when these nodes were added but the resulting graph structure?

Reply:
The implementation cannot tell the difference between two functionally identical graphs but with different edge or node ordering, for example:

```
NodeA        NodeC
  |             |
NodeB        NodeD
```
cannot be distinguished from
```
NodeC        NodeA
  |             |
NodeD        NodeB
```
If we cannot catch mistakes or collisions we cannot raise errors to let the user know they have made a mistake, so we have to put the onus on the user to not do this.

Reply:
I understand but my comment is more about clarifying to the reader what "in the same order" means. This might be especially confusing for the explicit API where one can add nodes and edges independent of each other and end up with identical graphs. Your example might also be helpful here.

Reply:
The clarification then would be something like:

> All nodes and edges must be added in the same order, and all Graph-related API calls and recorded queue submissions must occur in the same order.

Reply:
> Your example might also be helpful here.

I don't know if I agree with this, IMO this is straying into design doc territory (where I think it makes sense to talk about). I don't think we need to justify this here, what can the user do with this information?

Reply:
> All nodes and edges must be added in the same order, and all Graph-related API calls and recorded queue submissions must occur in the same order.

This reads much clearer to me

> I don't know if I agree with this, IMO this is straying into design doc territory (where I think it makes sense to talk about). I don't think we need to justify this here, what can the user do with this information?

I think is more than a design documentation though since it restricts the users. But I think the clarification suffices.